seisidd/devices/beamline.py

```diff
@@ -9,48 +9,281 @@
 from ophyd   import Component
 
 
-class Slits(MotorBundle):
-    """Control the four blades that form the beam"""
+class SlitUpstream(MotorBundle):
+    """Upstream slit that controls the four blades that form the beam"""
+    #   The slit PVs are not meaningful at all, the actual name will depend on set up
+    #   Here is the upstream slit PVs at 1-ID-E   ("motorPV.LVIO" for soft limit check) 
+    #   1ide1:m20
+    #   1ide1:m21
+    #   1ide1:m22
+    #   1ide1:m23
+    #   didn't found actual control in the scripts, manually adjusted??
 
-    left   = Component(EpicsMotor, "PV_left",   name='left'  )
-    right  = Component(EpicsMotor, "PV_right",  name='right' )
-    top    = Component(EpicsMotor, "PV_top",    name='top'   )
-    bottom = Component(EpicsMotor, "PV_bottom", name='bottom')
+    h_ib    =   Component(EpicsMotor, "PV_inboard",                 name='h_ib'     )   # horizontal in board
+    h_ob    =   Component(EpicsMotor, "PV_outboard",                name='h_ob'     )   # horizontal out board
+    h_size  =   Component(EpicsMotor, "1ide1:Kohzu_E_upHsize.VAL",  name='h_size'   )   # horizontal size !!!!!need to check this
+    v_tp    =   Component(EpicsMotor, "PV_top",                     name='v_tp'     )   # vertial top
+    v_bt    =   Component(EpicsMotor, "PV_bottom",                  name='v_bt'     )   # vertial bottom
+    v_size  =   Component(EpicsMotor, "1ide1:Kohzu_E_upVsize.VAL",  name='v_size'   )   # vertical size !!!!!need to check this
 
+class SlitDownstream(MotorBundle):
+    """Downstream slit that controls the four blades that form the beam"""
+    h_ib    =   Component(EpicsMotor, "PV_inboard",                 name='h_ib'     )
+    h_ob    =   Component(EpicsMotor, "PV_outboard",                name='h_ob'     )
+    h_size  =   Component(EpicsMotor, "1ide1:Kohzu_E_dnHsize.VAL",  name='h_size'   )   # horizontal size !!!!!need to check this
+    v_tp    =   Component(EpicsMotor, "PV_top",                     name='v_tp'     )
+    v_bt    =   Component(EpicsMotor, "PV_bottom",                  name='v_bt'     )
+    v_size  =   Component(EpicsMotor, "1ide1:Kohzu_E_upVsize.VAL",  name='v_size'   )   # vertical size !!!!!need to check this
 
-class Attenuator(MotorBundle):
+### Per Peter, only motion in the y direction is allowed to avoid accidents
+### consider comment out all the other PV accesses.
+class FocusLens1(MotorBundle):
```

Comment:
I will comment out these focus lenses since they shouldn't be accessed by users. 
I will leave out the controls in the y direction for switching to focus beams.